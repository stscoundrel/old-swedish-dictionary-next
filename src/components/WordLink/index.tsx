import { DictionaryEntry, DictionaryEntryDTO } from 'lib/models/dictionary'
import Link from 'next/link'
import styles from './WordLink.module.scss'

interface WordLinkProps{
  data: DictionaryEntry | DictionaryEntryDTO,
  useLowerCase: boolean,
}

export default function WordLink({ data: { slug, headword }, useLowerCase = true }: WordLinkProps) {
  return (
   <Link key={`link${slug}`} href={`/word/${slug}`} className={styles.link}>
      {useLowerCase ? headword.toLowerCase() : headword}
    </Link>
  )
}
