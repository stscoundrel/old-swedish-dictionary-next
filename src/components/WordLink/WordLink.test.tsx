import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import WordLink from './index'

const word = {
  headword: 'afköra',
  partOfSpeech: 'vb',
  grammaticalAspect: 'v.',
  definitions: [
    ' , vräka. hafwi wald han afköra  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.',
  ],
  alternativeForms: ['-bortdrifva'],
  slug: 'afkora',
}

describe('WordLink component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<WordLink data={word} />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<WordLink data={word} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Has correct label', () => {
    const tree = renderer.create(<WordLink data={word} />)
    const { root } = tree

    expect(root.findByType('a').children).toEqual([word.headword])
  })
})
