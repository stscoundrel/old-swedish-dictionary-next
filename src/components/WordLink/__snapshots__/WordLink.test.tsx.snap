// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordLink component Matches snapshot 1`] = `
<a
  className="link"
  href="/word/afkora"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  afk√∂ra
</a>
`;
